CriteoPublisherSdk/Sources/Feedback/CR_FeedbackConsentGuard.m

```diff
@@ -70,11 +70,27 @@ - (void)sendFeedbackBatch {
 /**
  * Checks if we are allowed to collect metrics regarding TCF v2 purposes:
  * - Purpose 1 - Store and/or access information on a device
- * - Purpose 7 - Measure ad performance
+ *
+ * The TCF entries that are considered are the following:
+ * IABTCF_PurposeConsents: User consent to purpose
+ * IABTCF_PublisherRestrictions{ID}: Publisher restrictions on purpose {ID}
+ * IABTCF_VendorConsents: Consent for vendors
+ * IABTCF_VendorLegitimateInterests: Legitimate interests for vendors
+ *
  * For more details: https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/
  */
 - (BOOL)hasFeedbackConsent {
-  return [_consent.gdpr isConsentGivenForPurpose:1] && [_consent.gdpr isConsentGivenForPurpose:7];
+  CR_Gdpr *gdpr = _consent.gdpr;
+  BOOL hasUserConsent = [gdpr isConsentGivenForPurpose:1];
+  CR_GdprTcfPublisherRestrictionType publisherRestriction =
+      [gdpr publisherRestrictionsForPurpose:1];
+  BOOL hasPublisherRestrictions =
+      publisherRestriction == CR_GdprTcfPublisherRestrictionTypeNotAllowed ||
+      publisherRestriction == CR_GdprTcfPublisherRestrictionTypeRequireLegitimateInterest;
+  BOOL hasVendorConsent = [gdpr isVendorConsentGiven];
+  BOOL hasVendorLegitimateInterest = [gdpr hasVendorLegitimateInterest];
+  return hasUserConsent && !hasPublisherRestrictions && hasVendorConsent &&
```

Comment:
Privacy question: if `requiredLegitimateInterest` is set to `true`, and the vendor has legitimate interest established, don't we have the right to store?

Reply:
Not with my understanding of our internal policy, which are written so the "Vendor" consent does not override "User" consent.